src/App.js

```diff
@@ -27,18 +27,23 @@ class App extends Component {
   render() {
     return (
       <>
-      <Header/>
+      
       <Router>
-        <Switch>
-          <Route exact path="/" component={Home} />
-          <Route path="/catindex" component={CatIndex} />
-          <Route path="/catshow" component={CatShow} />
-          <Route path="/catnew" component={CatNew} />
-          <Route path="/catedit" component={CatEdit} />
-          <Route component={NotFound} />
-        </Switch>
+        <Header/>
+          <Switch>
+            <Route exact path="/" component={Home} />
+            <Route path="/catindex" render={(props)=> <CatIndex cats={this.state.cats}/>} />
+            <Route path="/catshow/:id" render= {(props) => {
+              let id = props.match.params.id
+              let cat = this.state.cats.find((catObject)=> catObject.id == id)
+              return <CatShow cat={cat}/>
+            }} />
+            <Route path="/catnew" component={CatNew} />
+            <Route path="/catedit" component={CatEdit} />
+            <Route component={NotFound} />
+          </Switch>
+        <Footer/>
       </Router>
-      <Footer/>
       
       </>
```

Comment:
Router takes the place of the fragments. Not a big deal but something you want to replace soon.